evm/hyperliquid-block-helper.js

```diff
@@ -0,0 +1,182 @@
+        ],
+    };
+
+    const signature = await wallet._signTypedData(domain, { Agent: types.Agent }, phantomAgent);
+    const { ethers } = require('ethers');
+    const sig = ethers.utils.splitSignature(signature);
+
+    return { r: sig.r, s: sig.s, v: sig.v };
+}
+
+async function sendRequest(action, signature, nonce, isMainnet) {
+    const payload = { action, signature, nonce };
+    const endpoint = isMainnet ? 'https://api.hyperliquid.xyz/exchange' : 'https://api.hyperliquid-testnet.xyz/exchange';
+
+    const curlCommand = `curl -s -X POST "${endpoint}" \
+        -H "Content-Type: application/json" \
+        -H "User-Agent: Mozilla/5.0 (compatible; Hyperliquid-Block-Helper/1.0)" \
+        -d '${JSON.stringify(payload)}' \
+        --connect-timeout 15 \
+        --max-time 30`;
+
+    const { stdout, stderr } = await execAsync(curlCommand);
+
+    if (stderr && !stderr.includes('curl')) {
+        throw new Error(`curl stderr: ${stderr}`);
+    }
+
+    const result = JSON.parse(stdout);
+    return result;
+}
+
+async function switchBlockSize(privateKey, useBig, network = 'mainnet') {
+    validateParameters({ isNonEmptyString: { privateKey } });
+
+    if (!privateKey.startsWith('0x')) {
+        privateKey = '0x' + privateKey;
+    }
+
+    if (privateKey.length !== 66) {
+        throw new Error(`Invalid private key length: ${privateKey.length}`);
+    }
+
+    const wallet = new Wallet(privateKey);
+    const isMainnet = network.toLowerCase() === 'mainnet';
+
+    const action = { type: 'evmUserModify', usingBigBlocks: useBig };
+    const nonce = Date.now();
+    const signature = await signL1Action(wallet, action, null, nonce, isMainnet);
+    const result = await sendRequest(action, signature, nonce, isMainnet);
+
+    if (result.status === 'ok') {
+        return result;
+    } else {
+        throw new Error(`API error: ${result.response}`);
+    }
+}
+
+async function processCommand(config, chain, options) {
+    const { privateKey, blockSize } = options;
+
+    validateParameters({
+        isNonEmptyString: { privateKey },
+        isNonEmptyString: { blockSize },
+    });
+
+    if (blockSize !== 'big' && blockSize !== 'small') {
+        throw new Error('Block size must be "big" or "small"');
+    }
+
+    // Check if this is a Hyperliquid chain
+    const isHyperliquidChain = chain.name.toLowerCase().includes('hyperliquid') || chain.axelarId.toLowerCase().includes('hyperliquid');
+
+    if (!isHyperliquidChain) {
+        throw new Error(`Chain "${chain.name}" is not supported. This script only works on Hyperliquid chains.`);
+    }
+
+    const isMainnet = options.env === 'mainnet';
+    const network = isMainnet ? 'mainnet' : 'testnet';
+    const useBig = blockSize === 'big';
+
+    printInfo('Block size', blockSize.toUpperCase());
+    printInfo('Network', network);
+
+    try {
+        const result = await switchBlockSize(privateKey, useBig, network);
```

Comment:
```suggestion
        const result = await updateBlockSize(privateKey, useBig, network);
```
doesn't need to be a switch, might be setting the same value again